mongo/gridfs/bucket.go

```diff
@@ -465,23 +503,25 @@ func createIndexIfNotExists(ctx context.Context, iv mongo.IndexView, model mongo
 		_ = c.Close(ctx)
 	}()
 
+	modelKeysBytes, err := bson.Marshal(model.Keys)
```

Comment:
We should probably change the callers to pass `bsoncore.Document` instead of a `mongo.IndexModel` because we only look at the keys document anyway. Also, the callers can use `bsoncore` directly to build the keys doc instead of going through the `Marshal` codepath.

Reply:
Ah, that's true. We do need the `IndexModel` here. By using bsoncore directly, I meant that we could construct the index specs using bsoncore functions like `bsoncore.AppendDocumentStart` instead of constructing a `bson.D` which later gets marshalled to raw BSON anyway. I don't think this is a necessary change if we need the IndexModel anyway, though, because it'd look weird to do `model.Keys.([]byte)` and this isn't a critical code path anyway. Resolving.